provider/docker.go

```diff
@@ -128,6 +130,16 @@ func (provider *Docker) Provide(configurationChan chan<- types.ConfigMessage, po
 			ctx := context.Background()
 			version, err := dockerClient.ServerVersion(ctx)
 			log.Debugf("Docker connection established with docker %s (API %s)", version.Version, version.APIVersion)
+
+			hostname, err := os.Hostname()
+			if err != nil {
+				log.Errorf("Failed to get hostname for docker, error %s", err)
+				return err
+			}
+			if containerExists(ctx, dockerClient, hostname) {
+				provider.traefikContainerID = hostname
```

Comment:
Same here. If you run it in cluster mode, wouldn't that be overwritten quite often and not beeing "stable"?